.github/workflows/ci.yml

```diff
@@ -428,6 +461,84 @@ jobs:
           export PATH="$PWD/bin:$PWD/dist/bin:$PATH"
           dist-tests/${{ matrix.suite }}
 
+  coverage:
```

Comment:
Thanks for breaking this out, it feels less complicated this way than previously, at the cost of some duplication, but I would like your impression as well.

Also, please add a comment to indicate that this is similar to the `cabal-test` job but that it's different (because it runs only the single configuration that was hpc built and runs only the tests for which coverage is collected, uploading those and publishing them at the end).

Reply:
I agree. I think this is cleaner! I added a comment explaining the `coverage` job.